core/music/SpotifyApi.js

```diff
@@ -0,0 +1,112 @@
+class SpotifyApi {
+    constructor(client) {
+        this.client = client;
+        this.access_token = null;
+        this.client_id = this.client.config.player.key.spotifyClient;
+        this.client_secret = this.client.config.player.key.spotifySecret;
+    }
+
+    async fetchToken() {
+        const data = await fetch(`https://accounts.spotify.com/api/token?grant_type=client_credentials&client_id=${this.client_id}&client_secret=${this.client_secret}`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/x-www-form-urlencoded',
+            },
+        })
+            .then(res => res.json())
+            .catch(err => console.error(err));
+
+        this.access_token = data.access_token;
+
+        this.client.log({
+            textContent: 'Spotify token fetch',
+            headers: 'Spotify',
+            type: 'Success',
+        });
+
+        setTimeout(() => {
+            this.fetchToken();
```

Comment:
when client id and secret are not set this will send a request & a log every second as fetch doesn't throw error